k2/csrc/intersect_dense_pruned.cu

```diff
@@ -722,6 +828,30 @@ class MultiGraphDenseIntersectPruned {
     Ragged<ArcInfo> ai(ai_shape);
     ArcInfo *ai_data = ai.values.Data();  // uninitialized
 
+    // A valid final arc means its label == -1.
+    auto has_valid_final_arc = Array1<bool>(c_, NumFsas(), false);
+    bool *has_valid_final_arc_data = has_valid_final_arc.Data();
+    bool allow_partial = allow_partial_;
+
+    if (allow_partial_) {
```

Comment:
maybe meant to use allow_partial not allow_partial_ (or the variable was not needed?)